sdk/core/Azure.Core/tests/OperationTests.cs

```diff
@@ -133,5 +133,127 @@ public void OperationId()
+            var rehydrationToken = new RehydrationToken(null, null, "None", $"https://management.azure.com/subscriptions/subscription-id/providers/Microsoft.Compute/locations/region/operations/{operationId}?api-version=2019-12-01", "https://test", RequestMethod.Delete, null, OperationFinalStateVia.AzureAsyncOperation.ToString());
+            var operation = await Operation.RehydrateAsync<MockJsonModel>(pipeline, rehydrationToken);
+            Assert.NotNull(operation);
+            Assert.AreEqual(operationId, operation.Id);
+            Assert.True(operation.HasCompleted);
+            Assert.True(operation.HasValue);
+            Assert.AreEqual(200, operation.GetRawResponse().Status);
+            Assert.AreEqual(ModelReaderWriter.Write(mockJsonModel).ToString(), ModelReaderWriter.Write(operation.Value).ToString());
+        }
+
+        [Test]
+        public async Task ConstructOperationForRehydrationWithFailure()
+        {
+            HttpPipeline pipeline = CreateMockHttpPipelineWithFailure();
+            var operationId = Guid.NewGuid().ToString();
+            var rehydrationToken = new RehydrationToken(null, null, "None", $"https://management.azure.com/subscriptions/subscription-id/providers/Microsoft.Compute/locations/region/operations/{operationId}?api-version=2019-12-01", "https://test", RequestMethod.Delete, null, OperationFinalStateVia.AzureAsyncOperation.ToString());
+            var operation = await Operation.RehydrateAsync<MockJsonModel>(pipeline, rehydrationToken);
+            Assert.NotNull(operation);
+            Assert.AreEqual(500, operation.GetRawResponse().Status);
+            Assert.True(operation.HasCompleted);
+        }
+
+        [Test]
+        public async Task GetRehydrationTokenAsync()
+        {
+            var operationId = Guid.NewGuid().ToString();
+            var rehydrationToken = new RehydrationToken(operationId, null, "None", $"https://management.azure.com/subscriptions/subscription-id/providers/Microsoft.Compute/locations/region/operations/{operationId}?api-version=2019-12-01", "https://test/", RequestMethod.Delete, null, OperationFinalStateVia.AzureAsyncOperation.ToString());
+            var operation = await Operation.RehydrateAsync(HttpPipelineBuilder.Build(new MockClientOptions()), rehydrationToken);
+            var token = operation.GetRehydrationToken();
+            Assert.AreEqual(ModelReaderWriter.Write(rehydrationToken).ToString(), ModelReaderWriter.Write(token).ToString());
+        }
+
+        [Test]
+        public async Task GetRehydrationTokenOfTAsync()
+        {
+            var operationId = Guid.NewGuid().ToString();
+            var rehydrationToken = new RehydrationToken(operationId, null, "None", $"https://management.azure.com/subscriptions/subscription-id/providers/Microsoft.Compute/locations/region/operations/{operationId}?api-version=2019-12-01", "https://test/", RequestMethod.Delete, null, OperationFinalStateVia.AzureAsyncOperation.ToString());
```

Comment:
We should use a less real URL for this test.  The scanner has been more aggressively flagging anything that looks like it may be real.

Reply:
updated